lib/wibox/widget/slider.lua

```diff
@@ -505,6 +534,68 @@ function slider:fit(_, width, height)
     return width, height
 end
 
+-- @property handle_widget
+-- @tparam widget|nil handle_widget
+
+-- @property bar_widget
+-- @tparam widget|nil bar_widget
+function slider:layout(context, width, height)
+    local result = {}
+    local handle_widget = self._private.handle_widget
+        or nil
+    local bar_widget = self._private.bar_widget
+        or nil
+    --[[ if not handle_widget or not bar_widget then
+        return {}
+    end ]]
+
+
+    local value = self._private.value or self._private.min or 0
+    local min, _, interval = get_extremums(self)
+    local bar_height, bar_width = self._private.bar_height or height, width
+    local handle_height, handle_width = height, self._private.handle_width
+        or beautiful.slider_handle_width
+        or math.floor(height / 2)
```

Comment:
Is there a particular reason to make the width defaults to half the height?

I would argue we can make it a square and save this `math.floor` + division.

Reply:
I believe that was something I copied from the draw function. Eliminating the `math.floor` is definitely smarter, I'll be sure to fix that.